### ISSUE TITLE
BlenderBim : Removing Material from list removes all instances

### ISSUE THREAD

**Issue by Gorgious56** [IfcOpenShell/IfcOpenShell#1695]
It's possible to add the same material several times to a single material list. 
![image](https://user-images.githubusercontent.com/25156105/130785786-dd9b2637-8872-4d3d-816b-70276f6f3ff6.png)
However when clicking on the "Remove" icon on one of the items, all materials get removed and the list becomes empty.
![image](https://user-images.githubusercontent.com/25156105/130785857-d9dfab90-b325-4fbb-bd7f-4abb2dd2d277.png)
Either it shouldn't be possible to add the same material more than once to the list, or removing one instance of the material should only remove one item. Or am I missing something ?

Either way I can contribute on changing the behaviour if needed.

**Comment by Moult** [comment]
Looks like a bug on remove. Will investigate tomorrow.

**Comment by Moult** [comment]
Of course, don't let that dissuade you from investigating a fix yourself :)

**Comment by Gorgious56** [comment]
Haha, will do ! :)

At first glance it seems the code assumes the material can only be found once in the list, and it's just filtering it across the board.

https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.6.0/src/ifcopenshell-python/ifcopenshell/api/material/remove_list_item.py#L12-L13

Maybe using a list item index rather than the material id would prevent that ?

**Comment by Moult** [comment]
Yes - I just realised this is an `IfcMaterialList`.

An `IfcMaterialList` is a deprecated list of materials used in IFC2X3. We support it, but it isn't very pleasant. Looking at the docs, an IfcMaterialList indeed has a list of materials.

I would personally question this and say that it makes more sense to have a set of materials, not a list, but whatever, it's deprecated anyway.

So yes, feel free to use a list item index.

Or, if you reckon we should disobey the spec here (anyway, it's deprecated) and use a set of materials, you can edit this instead: https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.6.0/src/ifcopenshell-python/ifcopenshell/api/material/add_list_item.py#L12-L13
